src/TimeoutMigrationTool.ASQ.IntegrationTests/ASQTargetTests.cs

```diff
@@ -0,0 +1,378 @@
+{
+    using Microsoft.Azure.Cosmos.Table;
+    using NUnit.Framework;
+    using Particular.TimeoutMigrationTool;
+    using Particular.TimeoutMigrationTool.ASQ;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    class ASQTargetTests
+    {
+        string connectionString = Environment.GetEnvironmentVariable("AzureStorageQueue_ConnectionString") ?? "UseDevelopmentStorage=true;";
+        string endpointName;
+
+        [SetUp]
+        public void Setup()
+        {
+            endpointName = $"TestEndpoint{Guid.NewGuid().ToString().Replace("-", string.Empty)}";
+        }
+
+        [TearDown]
+        public async Task TearDown()
+        {
+            var nameProvider = new DelayedDeliveryTableNameProvider();
+
+            await DeleteTable(nameProvider.GetStagingTableName(endpointName)).ConfigureAwait(false);
+            await DeleteTable(nameProvider.GetDelayedDeliveryTableName(endpointName)).ConfigureAwait(false);
+        }
+
+        [Test]
+        public async Task AbleToMigrate_fails_if_Timeout_table_Does_not_exist()
+        {
+            // Arrange
+            var timeoutTarget = new ASQTarget(new TestLoggingAdapter(), connectionString, new DelayedDeliveryTableNameProvider("TimeoutTableThatDoesNotExist"));
+
+            // Act
+            var ableToMigrate = await timeoutTarget.AbleToMigrate(new EndpointInfo { EndpointName = endpointName }).ConfigureAwait(false);
+
+            // Assert
+            Assert.IsFalse(ableToMigrate.CanMigrate);
+            Assert.AreEqual("Target delayed delivery table TimeoutTableThatDoesNotExist does not exist.", ableToMigrate.Problems[0]);
+        }
+
+        [Test]
+        public async Task AbleToMigrate_fails_with_incorrect_connection_string()
+        {
+            // Arrange
+            var timeoutTarget = new ASQTarget(new TestLoggingAdapter(), "DefaultEndpointsProtocol=https;AccountName=fakename;AccountKey=g94OvNO9o3sVan5eipemQEHmU8zD2M9iq98E8nKSdR2bTuQB1hi07Yd1/8dDw6+1jGI2klWjpvoDahHPhR/3og==", new DelayedDeliveryTableNameProvider("TimeoutTableThatDoesNotExist"));
+
+            // Act
+            var ableToMigrate = await timeoutTarget.AbleToMigrate(new EndpointInfo { EndpointName = endpointName }).ConfigureAwait(true);
+
+            // Assert
+            Assert.IsFalse(ableToMigrate.CanMigrate);
+            Assert.AreEqual("Unable to connect to the storage instance with connection string 'DefaultEndpointsProtocol=https;AccountName=fakename;AccountKey=g94OvNO9o3sVan5eipemQEHmU8zD2M9iq98E8nKSdR2bTuQB1hi07Yd1/8dDw6+1jGI2klWjpvoDahHPhR/3og=='. Exception message 'No such host is known.'", ableToMigrate.Problems[0]);
+        }
+
+        [Test]
+        public async Task AbleToMigrate_passes_if_Timeout_table_exists()
+        {
+            // Arrange
+            await CreateTimeoutTable($"T{endpointName.Replace("-", string.Empty)}").ConfigureAwait(true);
```

Comment:
Why do you need to replace here again. Doesn't the endpoint name already have the `-` replaced?

Reply:
Good point.